diff --git a/core/misc/ajax.js b/core/misc/ajax.js
index 58759ac7a0..27272f2520 100644
--- a/core/misc/ajax.js
+++ b/core/misc/ajax.js
@@ -435,6 +435,8 @@ function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr
     }
 
     this.settings = null;
+    // Let other scripts act after Drupal process successful ajax response.
+    $(this.element).trigger('ajax.success');
   };
 
   Drupal.Ajax.prototype.getEffect = function (response) {
@@ -475,6 +477,8 @@ function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr
       var settings = this.settings || drupalSettings;
       Drupal.attachBehaviors(this.$form.get(0), settings);
     }
+    // Let other scripts act after Drupal process failing ajax response.
+    $(this.element).trigger('ajax.error');
     throw new Drupal.AjaxError(xmlhttprequest, uri, customMessage);
   };
 
diff --git a/core/modules/file/file.js b/core/modules/file/file.js
index 652028c60b..4e67eb07b4 100644
--- a/core/modules/file/file.js
+++ b/core/modules/file/file.js
@@ -90,7 +90,21 @@
       }
     },
     triggerUploadButton: function triggerUploadButton(event) {
-      $(event.target).closest('.js-form-managed-file').find('.js-form-submit').trigger('mousedown');
+      var $fileInput = $(event.target);
+      var $submitElement = $fileInput.closest('.js-form-managed-file').find('.js-form-submit')
+      $submitElement.prop('ajax-uploading-file-id', $fileInput.attr('id'));
+      // Handle errors from outside Drupal (like web server or network)
+      $submitElement.once('file-ajax-error-handling').bind('ajax.error', function() {
+        var $this = $(this);
+        $fileInput = $('#' + $this.prop('ajax-uploading-file-id'));
+        if ($fileInput.length) {
+          // Empty input and show error message
+          $fileInput.val('');
+          $(this).closest('div.js-form-managed-file').prepend('<div class="messages messages--error file-upload-js-error" aria-live="polite">' +
+            'An error occurred while uploading the file. Please check your file and try again.' + '</div>');
+        }
+      });
+      $submitElement.trigger('mousedown');
     },
     disableFields: function disableFields(event) {
       var $clickedButton = $(this);
